src/dune_engine/action.mli

```diff
@@ -172,5 +172,8 @@ module Full : sig
   val add_sandbox : Sandbox_config.t -> t -> t
   val add_can_go_in_shared_cache : bool -> t -> t
 
+  (** Marks action as safe to be cached *)
+  val allowed_in_shared_cache : t -> t
```

Comment:
I'd say just use the existing function for now.

Reply:
Removed it again as it wasn't needed after all.